### DISCUSSION TITLE
Append Uses Default Value

### DISCUSSION THREAD

**Discussion by sethdumaguin** [react-hook-form/react-hook-form#8364] [General]
Hi!

So I have a useFieldArray form with defaultValues. When I remove a field and then append, the appended input still uses the defaultValue, but I would want it to be empty. Is there a way for the field array inputs to not use the defaultValue anymore once they are removed?

Appreciate the help!

See example below:
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-t1xl9o?file=/src/index.js

**Comment by Moshyfawn** [comment]
You should provide the default input's value when appending the field: an empty string for a string input, for example.

From [RHF useFieldArray docs](https://react-hook-form.com/api/usefieldarray):

![Screenshot_20220517_104559.jpg](https://user-images.githubusercontent.com/16290753/168757832-6d0debb1-f024-4564-9ea5-aa4c10b51c50.jpg)

**Comment by athulmurali** [comment]
- Looking into the source code to see if we can make it  controlled by a flag  options 
- on empty loads rows with undefined or empty 
append () -> appends with (field1...field n : undefined on each)
